### PR TITLE
*: Make go vet --shadow happy

### PR DESCRIPTION
**Pull request by ngaut**
Also do some clean up.

### CODE REVIEW

File: ddl/index.go [hunk 1]
```diff
@@ -478,10 +478,8 @@ func (d *ddl) dropTableIndex(t table.Table, indexInfo *model.IndexInfo) error {
 	prefix := kv.GenIndexPrefix(t.IndexPrefix(), indexInfo.Name.L)
 	prefixBytes := []byte(prefix)
 
-	keys := make([]string, maxBatchSize)
-
 	for {
-		keys := keys[0:0]
+		keys := make([]string, 0, maxBatchSize)
```
Comment (siddontang) [inline ddl/index.go]: I just want to reuse the keys buffer, no need to create a new one every time.

Comment (ngaut) [inline ddl/index.go]: Since it's already batched, it wouldn't be a problem. Let's make it clean.

**Comment by siddontang** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**